cmd/soroban-rpc/internal/daemon/daemon.go

```diff
@@ -275,19 +276,25 @@ func (d *Daemon) Run() {
 		"addr":    d.server.Addr,
 	}).Info("starting Soroban JSON RPC server")
 
-	go func() {
+	util.MonitoredRoutine(util.MonitoredRoutineConfiguration{
+		Log:                d.logger,
+		ExitProcessOnPanic: true,
```

Comment:
do you intend to include the PanicCounter also? that's a nice feature of this new utility, to get that into metrics.

Reply:
it conflicts with ExitProcessOnPanic.. so it would be one or the other.